docs/src/modules/java/pages/workflow-entities.adoc

```diff
@@ -96,16 +96,77 @@ include::example$java-spring-transfer-workflow/src/main/java/com/example/transfe
+
+[source,java,indent=0]
+.src/main/java/com/example/transfer/TransferWorkflow.java
+----
+include::example$java-spring-transfer-workflow-compensation/src/main/java/com/example/transfer/TransferWorkflow.java[tag=timeouts]
+----
+<1> Sets a workflow global timeout.
+<2> Sets a default timeout for all workflow steps.
+
+A default step timeout can be overridden in step builder.
+
+[source,java,indent=0]
+.src/main/java/com/example/transfer/TransferWorkflow.java
+----
+include::example$java-spring-transfer-workflow-compensation/src/main/java/com/example/transfer/TransferWorkflow.java[tag=step-timeout]
+----
+<1> Overrides a step timeout for a specific step.
+
+=== Recover strategy
+
+It's time to define what should happen in case of timeout or any other unhandled error.
+
+[source,java,indent=0]
+.src/main/java/com/example/transfer/TransferWorkflow.java
+----
+include::example$java-spring-transfer-workflow-compensation/src/main/java/com/example/transfer/TransferWorkflow.java[tag=recover-strategy]
+----
+<1> Sets a failover transition in case of a workflow timeout.
+<2> Sets a default failover transition for all steps with maximum number of retries.
+<3> Overrides a step recovery strategy only for `deposit` step.
+
+
+NOTE: In case of a workflow timeout one last failover step can be performed. Transitions from that failover step will be ignored.
+
+=== Compensation
+
+The idea behind the Workflow Entity error handling is that workflows should only fail due to unknown errors during execution. In general, you should always write your workflows so that they do not fail on any known edge cases. If you expect an error, it's better to be explicit about it, possibly with your domain types. Based on this information and the flexible Workflow Entity API you can define a compensation for any workflow step.
+
+[source,java,indent=0]
+.src/main/java/com/example/transfer/TransferWorkflow.java
+----
+include::example$java-spring-transfer-workflow-compensation/src/main/java/com/example/transfer/TransferWorkflow.java[tag=compensation]
+----
+<1> Explicit deposit call result type.
+<2> Finishes workflow as completed, in the case of a successful deposit.
+<3> Launches compensation step to handle deposit failure. A withdraw step must be reversed.
+<4> Compensation step is just any other step, with the same set of functionalities.
+<5> Correct compensation can finish the workflow.
+<6> Any other result might be handled by a default recovery strategy.
+
+Compensating a workflow step (steps) might involve multiple logical steps and thus is part of the overall business logic that must be defined within the workflow itself. For simplicity, in the example above, the compensation is applied only to `withdraw` step. Whereas `deposit` step itself might also require a compensation. In case of a step timeout we can't be certain about step successful or error outcome.
+
+//TODO fix url
```

Comment:
which url?

Reply:
the one below, I will fix it after I merge this change since this repo does not exist yet.